### PR TITLE
feat: soil ID slice/service

### LINKED ISSUE #10: Create soil data input redux slice/service with phase 1 data inputs
## Description
See the user stories tracking this issue for which data inputs to include. Depends on https://github.com/techmatters/terraso-backend/issues/622

### LINKED ISSUE #11: Create soil project redux slice/service with ability to edit depth intervals
## Description
See https://github.com/techmatters/terraso-backend/issues/622 for associated backend issue.

### LINKED ISSUE #85: add data collection methods to project redux slice/service

### PR DESCRIPTION
**Pull request by shrouxm**
## Description
Creates a soil ID redux slice and service exposing all of the new soil ID related endpoints created in techmatters/terraso-backend#843 and techmatters/terraso-backend#819

### Checklist
- [x] Corresponding issue has been opened

Fixes #10
Fixes #11 
Fixes #85

### CODE REVIEW

**Comment by shrouxm** [comment]
@david-code could you look this over some time soon? 

i couldn't figure out how to fit the dehydrated/collapseFields pattern into the query i needed to make to fetch all the soil data related site/project stuff so i had to mess with it a lot. the old `dehydrated` function is replaced by `dispatchByKeys`, and `collapseFields` has kinda gotten broken down into different helper methods that get composed differently in each service. i am also experimenting with defining the client data model directly from the graphql schema in `soilIdSlice` since copying out that truly horrendous blob of fields felt quite unappealing!

i will try to use these changes with the frontend tomorrow to make sure it doesn't somehow not work in a terrible way

**Review by david-code — COMMENTED** [review]
Nice, looks really good! I will take another look when you move it out from draft, but looks great so far.

**Comment by shrouxm** [comment]
@david-code ok this is ready for final review!

**Review by david-code — APPROVED** [review]

File: src/soilId/soilIdSlice.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+const initialState = {
+  soilData: {} as Record<string, SoilData>,
+  projectSettings: {} as Record<string, ProjectSoilSettings>,
+};
+
+export const sameDepth =
+  (a: { depthInterval: DepthInterval }) =>
+  (b: { depthInterval: DepthInterval }) =>
+    a.depthInterval.start === b.depthInterval.start &&
+    a.depthInterval.end === b.depthInterval.end;
```
[the remark this answers is not in the document]
Comment (shrouxm) [reply src/soilId/soilIdSlice.ts:98]: not until we need one! i'm not sure if we'll ever actually be doing any computation that would require comparison

**Pull request merged**